hs_core/views/__init__.py

```diff
@@ -2238,6 +2238,61 @@ def hsapi_get_user(request, user_identifier):
     return get_user_or_group_data(request, user_identifier, "false")
 
 
+@swagger_auto_schema(
+    method="post",
+    operation_description="Check user password for Keycloak Migration",
+    responses={200: "Password is valid", 400: "Password is invalid"},
+    manual_parameters=[uid],
+    request_body=openapi.Schema(type=openapi.TYPE_OBJECT,
+                                properties={'password': openapi.Schema(type=openapi.TYPE_STRING,
+                                                                       description="raw password to validate")}
+                                )
+)
+@swagger_auto_schema(
+    method="get",
+    operation_description="Get user data for Keycloak Migration",
+    responses={200: "Returns JsonResponse containing user data"},
+    manual_parameters=[uid],
+)
+@api_view(["GET", "POST"])
+def hsapi_get_user_for_keycloak(request, user_identifier):
+    """
+    Get user data
+
+    :param user_identifier: id of the user for which data is needed
+    :return: JsonResponse containing user data
+    """    
+    user: User = hydroshare.utils.user_from_id(user_identifier)
+
+    if not user.is_superuser:
```

Comment:
@sblack-usu should we be checking if `request.user` is admin, not whether the user_identifier is that of an admin user?

Reply:
@sblack-usu I think I don't really understand how this endpoint is used by Keycloak.
Is KC calling this endpoint to get a single admin user?